### DISCUSSION TITLE
Carla 0.9.14 Build Error on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by rreddy78** [carla-simulator/carla#6144] [Q&A] [answered]
I think I have followed the build instructions but I get this error:

CARLA version: Release branch of CARLA
Platform/OS: Ubuntu 20.04
Problem you have experienced: make launch and make package is failing
What you expected to happen: build should have worked
Steps to reproduce: Latest CARLA master branch
Other information (documentation you consulted, workarounds you tried): Latest UnrealEngine (carla branch).
I am following the instructions of the documentation: https://carla.readthedocs.io/en/latest/build_linux/#unreal-engine

UnrealEngine build succeeded.
make PythonAPI succeeded

make launch gave me this error first:

Time Elapsed 00:00:01.98
Log file: /home/sxv1kor/Downloads/GitHub/UnrealEngine/Engine/Programs/UnrealBuildTool/Log_GPF.txt
Some Platforms were skipped due to invalid SDK setup: Android.
See the log file for detailed information
Discovering modules, targets and source code for project...
/home/sxv1kor/Downloads/carla/Unreal/CarlaUE4/Plugins/CarlaTools/Source/CarlaTools/CarlaTools.Build.cs(11,104): error CS0117: 'UnrealTargetPlatform' does not contain a definition for 'Win32'
/home/sxv1kor/Downloads/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Carla.Build.cs(14,104): error CS0117: 'UnrealTargetPlatform' does not contain a definition for 'Win32'
Expecting to find a type to be declared in a target rules named 'CarlaUE4Target'. This type must derive from the 'TargetRules' type defined by Unreal Build Tool.
BuildCarlaUE4.sh: Build C

**Comment by rreddy78** [comment, marked as answer]
I see now why I get this error. I have not followed the build instructions..
Have used the UnrealEngine 5.1 (carla branch) instead of the fork for Carla 4.26 as given in docs!

**Reply by jjletsgo** [reply]
can you teach me how to use fork for Carla 4.26?????

**Comment by SushilaRatre** [comment]
C:\carla>make launch
-[Setup]: [Batch params]: --boost-toolset msvc-14.2 --all
-[Setup]: Asynchronous jobs:  16
-[Setup]: Boost toolset:      msvc-14.2
-[Setup]: Install directory:  "C:\carla\Build\"
-[Setup]: Installing zlib...
    -[install_zlib]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_zlib]: A zlib installation already exists.
    -[install_zlib]: Delete "C:\carla\Build\zlib-install\" if you want to force a rebuild.
    -[install_zlib]: Exiting...
-[Setup]: Installing libpng...
    -[install_libpng]: [Batch params]: --build-dir "C:\carla\Build\" --zlib-install-dir "C:\carla\Build\zlib-install\"
    -[install_libpng]: A libpng installation already exists.
    -[install_libpng]: Delete "C:\carla\Build\libpng-1.2.37-install\" if you want to force a rebuild.
    -[install_libpng]: Exiting...
-[Setup]: Installing rpclib...
    -[install_rpclib]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_rpclib]: A rpclib installation already exists.
    -[install_rpclib]: Delete "C:\carla\Build\rpclib-install\" if you want to force a rebuild.
    -[install_rpclib]: Exiting...
-[Setup]: Installing Google Test...
    -[install_gtest]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_gtest]: A Google Test installation already exists.
    -[install_gtest]: Delete "C:\carla\Build\gtest-install\" if you want to force a rebuild.
    -[install_gtest]: Exiting...
-[Setup]: Installing "Recast & Detour"...
    -[install_recast]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_recast]: A "Recast & Detour" installation already exists.
    -[install_recast]: Delete "C:\carla\Build\recast-0b13b0-install\" if you want to force a rebuild.
    -[install_recast]: Exiting...
-[Setup]: Installing Boost...
    -[install_boost]: [Batch params]: --build-dir "C:\carla\Build\" --toolset msvc-14.2 --version 1.80.0 -j 16
    -[install_boost]: A Boost installation already exists.
    -[install_boost]: Delete "C:\carla\Build\boost-1.80.0-install\" if you want to force a rebuild.
    -[install_boost]: Exiting...
-[Setup]: Installing Xercesc...
    -[install_xercesc]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_xercesc]: A xerces installation already exists.
    -[install_xercesc]: Delete "C:\carla\Build\xerces-c-3.2.3-install\" if you want to force a rebuild.
    -[install_xercesc]: Exiting...
        1 file(s) copied.
-[Setup]: Installing Sqlite3
    -[install_sqlite3]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_sqlite3]: A Sqlite3 installation already exists.
    -[install_sqlite3]: Delete "C:\carla\Build\sqlite3-install" if you want to force a rebuild.
    -[install_sqlite3]: Exiting...
        1 file(s) copied.
-[Setup]: Installing PROJ
    -[install_proj]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_proj]: A PROJ installation already exists.
    -[install_proj]: Delete "C:\carla\Build\proj-install" if you want to force a rebuild.
    -[install_proj]: Exiting...
        1 file(s) copied.
-[Setup]: Installing Eigen
    -[install_eigen]: [Batch params]: --build-dir "C:\carla\Build\"
0 File(s) copied
0 File(s) copied
        1 file(s) copied.
        1 file(s) copied.

    -[install_eigen]: Eigen has been successfully installed in "C:\carla\Build\eigen-install"!
    -[install_eigen]: Exiting...
-[Setup]: Creating "CMakeLists.txt.in"...
-[Setup]:
   ###########
   # SUCCESS #
   ###########

   IMPORTANT!

   All the CARLA library dependences should be installed now.
   (You can remove all "*-src" folders in C:\carla\Build\ directory)

   You only need the ASSET PACK with all the meshes and textures.

   This script provides the assets for CARLA 0.9.14:
   You can download the assets from here:

       http://carla-assets.s3.amazonaws.com/20221201_5ec9328.tar.gz

   Unzip it in the "C:\carla\Unreal\CarlaUE4\Content\Carla\" folder.
   If you want another version, search it in C:\carla\Util\ContentVersions.txt.

-[BuildLibCarla]: [Batch params]: --server --client
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Build debug:   ON
-- Build release: ON
-- Build test:    ON
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: C:/carla/Build/libcarla-visualstudio
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  1>Checking Build System
  carla_server.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\server\Release\carla_server.lib
  carla_server_debug.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\server\Release\carla_server_debug.lib
  1>
  -- Install configuration: "Release"
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Build debug:   ON
-- Build release: ON
-- Build test:    ON
-- Configuring done (0.1s)
-- Generating done (0.1s)
-- Build files have been written to: C:/carla/Build/libcarla-visualstudio
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  1>Checking Build System
  carla_client.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\client\Release\carla_client.lib
  carla_client_debug.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\client\Release\carla_client_debug.lib
  1>
  -- Install configuration: "Release"
-[BuildLibCarla]: LibCarla server has been successfully installed in "C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\"!
-[BuildLibCarla]: LibCarla client has been successfully installed in "C:\carla\PythonAPI\carla\dependencies\"!
-[BuildCarlaUE4]: [Batch params]: --build --all
--build
--all
ECHO is off.
-[BuildCarlaUE4]: Building Unreal Editor...
Using bundled DotNet SDK version: 6.0.302
Running UnrealBuildTool: dotnet "..\..\Engine\Binaries\DotNET\UnrealBuildTool\UnrealBuildTool.dll" CarlaUE4Editor        Win64        Development        -WaitMutex        -FromMsBuild        "C:/carla/Unreal/CarlaUE4/CarlaUE4.uproject"
Log file: C:\Users\TEJAS UMESH RATRE\AppData\Local\UnrealBuildTool\Log.txt
**Creating makefile for CarlaUE4Editor (no existing makefile)
C:\carla\Unreal\CarlaUE4\Plugins\CarlaTools\Source\CarlaTools\CarlaTools.Build.cs(11,104): error CS0117: 'UnrealTargetPlatform' does not contain a definition for 'Win32'
C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Carla.Build.cs(14,104): error CS0117: 'UnrealTargetPlatform' does not contain a definition for 'Win32'
Expecting to find a type to be declared in a target rules named 'CarlaUE4EditorTarget'.  This type must derive from the 'TargetRules' type defined by Unreal Build Tool.
make: *** [CarlaUE4Editor] Error 6**


Hi, I am also getting the above error at the time of make launch in windows.
Kindly let me know the solution. Thanks.
